src/components/App/App.js

```diff
@@ -3,26 +3,56 @@ import React, { Component } from "react";
 import Header from "../Header/Header";
 import MovieContainer from "../MovieContainer/MovieContainer";
 import movieData from "../../mockData/movieData";
+import Hero from "../Hero/Hero";
+import { fetchMovieData } from "../../apiCalls";
 
 class App extends Component {
   constructor() {
     super();
     this.state = {
-      movies: movieData.movies,
+      allMovies: movieData.movies,
+      movies: [],
+      movieForHero: {},
+      error: "",
+      isLoading: true,
     };
   }
 
-  randomMoviePoster = () => {
-    const index = Math.floor(Math.random() * this.state.movies.length);
-    const randomMovie = this.state.movies[index];
-    return randomMovie;
+  generateRandomMoviePoster = () => {
+    const index = Math.floor(Math.random() * this.state.allMovies.length);
+    const randomMovie = this.state.allMovies[index];
+    this.setState({
+      movieForHero: randomMovie,
+      isLoading: false,
+    });
+  };
+
+  componentDidMount = () => {
+    fetchMovieData("")
+      .then((data) => {
+        this.setState({ movies: data.movies});
+        this.generateRandomMoviePoster();
+      })
+      .catch((err) => {
+        console.log(`${err.name}, ${err.message}`);
+        this.setState({ error: "Sorry, something went wrong" });
+      });
   };
 
   render() {
+    if (this.state.isLoading) {
+      return <p>Loading...</p>;
+    }
     return (
       <div className="App">
-        <Header randomMovie={this.randomMoviePoster()} />
-        <MovieContainer movieData={this.state.movies} />
+        {/* {this.state.error && <h2>{this.state.error}</h2>} */}
```

Comment:
I see this is commented out now.  Eager to see where/how you end up displaying the error if your app encounters one.